server/issue.go

```diff
@@ -193,6 +193,75 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
 	return http.StatusOK, nil
 }
 
+func (p *Plugin) assignJiraIssue(mmUserId, issueKey, assignee string) (string, error) {
+	ji, err := p.LoadCurrentJIRAInstance()
+	if err != nil {
+		return "", err
+	}
+
+	jiraUser, err := ji.GetPlugin().LoadJIRAUser(ji, mmUserId)
+	if err != nil {
+		return "", err
+	}
+
+	jiraClient, err := ji.GetJIRAClient(jiraUser)
+	if err != nil {
+		return "", err
+	}
+
+	// check for valid issue key
+	_, _, err = jiraClient.Issue.Get(issueKey, nil)
+	if err != nil {
+		errorMsg := fmt.Sprintf("We couldn't find the issue key `%s`.  Please confirm the issue key and try again.", issueKey)
+		return errorMsg, nil
+	}
+
+	// Get list of assignable assignees
+	url := fmt.Sprintf("rest/api/3/user/assignable/search?issueKey=%s&query=%s", issueKey, assignee)
+	req, err := jiraClient.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	res, err := jiraClient.Do(req, nil)
+	if err != nil {
+		if res.Response.StatusCode == 401 {
+			return "You do not have the appropriate permissions to perform this action. Please contact your Jira administrator.", nil
+		}
+		return "", err
+	}
+
+	var data []*jira.User
+	if err := json.NewDecoder(res.Body).Decode(&data); err != nil {
+		return "", errors.WithMessage(err, "failed to decode incoming request")
+	}
+	defer res.Body.Close()
```

Comment:
the defer should go before you read from res.Body (that way if something happens and you error out, the body will still be closed)

Reply:
Thanks for the added explanation to the request.  Adding and will commit asap